### PR TITLE
Flaky E2E: update the Calypso SSR spec.

### LINKED ISSUE #78186: Flaky E2E: infrastructure SSR specs.
### Spec file

test/e2e/specs/infrastructure/infrastructure__ssr-enabled.ts

### TeamCity ID

10206047

### Logs

```
page.goto: Timeout 10000ms exceeded.
=========================== logs ===========================
navigating to "https://container-priceless-curran.calypso.live/theme/twentytwenty", waiting until "domcontentloaded"
============================================================
at /home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/test/e2e/specs/infrastructure/infrastructure__ssr-enabled.ts:24:14
    at Object.<anonymous> (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-each/build/bind.js:79:13)
    at Promise.then.completed (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/utils.js:391:28)
    at callAsyncCircusFn (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/utils.js:316:10)
    at _callCircusTest (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/run.js:218:40)
    at _runTest (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/run.js:155:3)
    at _runTestsForDescribeBlock (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/run.js:66:9)
    at _runTestsForDescribeBlock (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/run.js:60:9)
    at run (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/run.js:25:3)
    at runAndTransformResultsToJestFormat (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapterInit.js:170:21)
    at jestAdapter (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapter.js:82:19)
    at runTestInternal (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-runner/build/runTest.js:389:16)
    at runTest (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-runner/build/runTest.js:475:34)
    at Object.worker (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-runner/build/testWorker.js:133:12)
```

### PR DESCRIPTION
**Pull request by worldomonation**
Closes https://github.com/Automattic/wp-calypso/issues/78186.

## Proposed Changes

This PR updates the Calypso SSR spec to be less flaky.

Key changes:
- each page now loads in its own tab.
- test step title is fixed instead of being variable with the container name.

Still unknown:
- whether these tests are necessary or even useful. Thread: p1686819783199949-slack-C7YPUHBB2

## Testing Instructions

Ensure the following build configurations are passing:
  - [x] Calypso E2E (desktop)
  - [x] Calypso E2E (mobile)

Revised spec managed to pass 100 iterations.
image

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: test/e2e/specs/infrastructure/infrastructure__ssr-enabled.ts [hunk 1]
```diff
@@ -1,27 +1,26 @@
 /**
- * @group calypso-pr
- * @group kpi
+ * @group calypso-loop
```
Comment (karenroldan) [inline test/e2e/specs/infrastructure/infrastructure__ssr-enabled.ts]: **question:** Is changing the group here intended? I don't see `calypso-loop` being used in our CI. I could be wrong though!

Comment (worldomonation) [reply test/e2e/specs/infrastructure/infrastructure__ssr-enabled.ts]: @karenroldan Yeah, that's a leftover from me running a stress loop. Sorry!

Comment (worldomonation) [reply test/e2e/specs/infrastructure/infrastructure__ssr-enabled.ts]: Fixed in b32ec86354e6bdd14c9e68b95848d52ce16436ea.

File: test/e2e/specs/infrastructure/infrastructure__ssr-enabled.ts [hunk 2]
```diff
@@ -1,27 +1,26 @@
 /**
  * @group calypso-pr
- * @group kpi
```
Comment (karenroldan) [inline test/e2e/specs/infrastructure/infrastructure__ssr-enabled.ts:3]: question (non-blocking): sorry, I missed this! Do we also need to remove this `kpi` group?

**Review by karenroldan — APPROVED** [review]
My bad for not approving yesterday! LGTM 🚀

File: test/e2e/specs/infrastructure/infrastructure__ssr-enabled.ts (hunk 2 above)
Comment (worldomonation) [reply test/e2e/specs/infrastructure/infrastructure__ssr-enabled.ts:3]: Yep, this was left over from half a year ago when I was working on a dashboard, so it's actually been an unintentional change all along.

**Pull request merged**